viam-cartographer.go

```diff
@@ -767,222 +679,57 @@ func (slamSvc *builtIn) StartSLAMProcess(ctx context.Context) error {
-
-	for i := 0; i < 2; i++ {
-		slamSvc.activeBackgroundWorkers.Add(1)
-		wg.Add(1)
-		if err := ctx.Err(); err != nil {
-			if !errors.Is(err, context.Canceled) {
-				slamSvc.logger.Errorw("unexpected error in SLAM service", "error", err)
-			}
-			slamSvc.activeBackgroundWorkers.Done()
-			return images, releaseFuncs, err
-		}
-		iLoop := i
-		goutils.PanicCapturingGo(func() {
-			defer slamSvc.activeBackgroundWorkers.Done()
-			defer wg.Done()
-			images[iLoop], releaseFuncs[iLoop], errs[iLoop] = slamSvc.getPNGImage(ctx, cams[iLoop])
-		})
-	}
-	wg.Wait()
-
-	for _, err := range errs {
-		if err != nil {
-			return images, releaseFuncs, err
-		}
-	}
-
-	return images, releaseFuncs, nil
-}
-
-// getAndSaveDataDense implements the data extraction for dense algos and saving to the directory path (data subfolder) specified in
-// the config. It returns the full filepath for each file saved along with any error associated with the data creation or saving.
-func (slamSvc *builtIn) getAndSaveDataDense(ctx context.Context, cams []camera.Camera) (string, error) {
-	ctx, span := trace.StartSpan(ctx, "slam::builtIn::getAndSaveDataDense")
+func (cartoSvc *cartographerService) getAndSaveData(ctx context.Context, cams []camera.Camera) (string, error) {
+	ctx, span := trace.StartSpan(ctx, "viamcartographer::cartographerService::getAndSaveData")
 	defer span.End()
 
 	if len(cams) != 1 {
-		return "", errors.Errorf("expected 1 camera for this slam algorithm, found %v", len(cams))
+		return "", errors.Errorf("expected one lidar camera for cartographer, found %v", len(cams))
 	}
 
 	pointcloud, err := cams[0].NextPointCloud(ctx)
 	if err != nil {
 		if err.Error() == opTimeoutErrorMessage {
-			slamSvc.logger.Warnw("Skipping this scan due to error", "error", err)
+			cartoSvc.logger.Warnw("Skipping this scan due to error", "error", err)
 			return "", nil
 		}
 		return "", err
 	}
 
-	var fileType string
-	switch slamSvc.slamMode {
-	case slam.Dim2d:
-		fileType = ".pcd"
-	case slam.Dim3d:
-		return "", errors.New("Dim3d is not implemented")
-	case slam.Rgbd, slam.Mono:
-		return "", errors.Errorf("bad slamMode %v specified for this algorithm", slamSvc.slamMode)
-	}
-	filenames, err := createTimestampFilenames(slamSvc.dataDirectory, slamSvc.primarySensorName, fileType, slamSvc.slamMode)
-	if err != nil {
-		return "", err
-	}
-	filename := filenames[0]
+	fileType := ".pcd"
+	filename := createTimestampFilenames(cartoSvc.dataDirectory, cartoSvc.primarySensorName, fileType)
 	return filename, dataprocess.WritePCDToFile(pointcloud, filename)
 }
 
 // Creates a file for camera data with the specified sensor name and timestamp written into the filename.
 // For RGBD cameras, two filenames are created with the same timestamp in different directories.
-func createTimestampFilenames(dataDirectory, primarySensorName, fileType string, slamMode slam.Mode) ([]string, error) {
+func createTimestampFilenames(dataDirectory, primarySensorName, fileType string) string {
```

Comment:
[nit] This creates a single filename, so should it be createTimestampFilename?

Reply:
Yes, good catch!